File: WindowsGrep/WindowsGrep.Common/Data/FileItem.cs
namespace WindowsGrep.Common;
public struct FileItem
{
    #region Properties..
    public string Name { get; private set; }
    public bool IsDirectory { get; private set; }
    public long FileSize { get; private set; }
    #endregion Properties..

    #region Constructors..
    public FileItem(string name, bool isDirectory, long fileSize)
    {
        Name = name;
        IsDirectory = isDirectory;
        FileSize = fileSize;
    }
    #endregion Constructors..
}

File: WindowsGrep/src/models/file_item.rs
#[derive(Debug, Clone)]
pub struct FileItem {
    pub name: String,
    pub is_directory: bool,
    pub file_size: i64,
}

impl FileItem {
    pub fn new(name: String, is_directory: bool, file_size: i64) -> Self {
        FileItem { name, is_directory, file_size }
    }
}
